core/include/detray/core/detector.hpp

```diff
@@ -604,6 +611,39 @@ class detector {
     bfield_type _bfield;
 };
 
+/// @brief A static inplementation of detector data for device
+template <typename metadata, template <typename> class bfield_t,
+          typename container_t>
+struct detector_buffer {
+
+    using detector_type = detector<metadata, bfield_t, container_t>;
+
+    detector_buffer(detector_type &det, vecmem::memory_resource &mr,
+                    vecmem::copy &cpy)
+        : _detector_buffer(
+              detray::get_buffer(detray::get_data(det.volumes()), mr, cpy),
+              detray::get_buffer(det.transform_store(), mr, cpy),
+              detray::get_buffer(det.mask_store(), mr, cpy),
+              detray::get_buffer(det.material_store(), mr, cpy),
+              detray::get_buffer(det.surface_store(), mr, cpy),
+              detray::get_buffer(det.volume_search_grid(), mr, cpy)),
+          _bfield_view(det.get_bfield()) {}
+
+    /*detector_buffer(detector_type &det, vecmem::memory_resource& mr,
+       vecmem::copy& cpy) : _detector_buffer(
+          detray::get_buffer(detray::get_data(det.volumes()), mr, cpy),
+          detray::get_buffer(det.transform_store(), mr, cpy),
+          detray::get_buffer(det.mask_store(), mr, cpy),
+          detray::get_buffer(det.material_store(), mr, cpy),
+          detray::get_buffer(det.surface_store(), mr, cpy),
+          detray::get_buffer(det.volume_search_grid(), mr, cpy)),
+          _bfield_view(det.get_bfield()) {}*/
+
+    // members
+    typename detector_type::buffer_type _detector_buffer;
+    typename detector_type::bfield_type::view_t _bfield_view;
```

Comment:
I don't know much about covfie but I think the detector buffer type shouldn't hold a view but an actual field type which would be setup with the correct backend (e.g. cuda). And then you can construct this from the host detector type with a simple copy constructor, like in this example
https://github.com/acts-project/covfie/blob/262b470216bd5799bff74379e67f85126a1bc447/examples/cuda/render_slice.cu#L122-L127

Reply:
Sounds good. I thought it was safer to copy the view around, but honestly I have no clue :) I think though that a rigorous handling of covfie and asynchronous copies/cuda streams etc. should go into a separate PR.

Reply:
I'm sure @stephenswat can give some input here but I think if you do that you might just be accessing data which will always be held by the host